From 3e605e95ef41d6ad60a484498ef54c713f1271d1 Mon Sep 17 00:00:00 2001
From: Sebastian Reichel <sre@kernel.org>
Date: Sat, 27 Dec 2025 17:31:50 +0100
Subject: [PATCH 104/263] dt-bindings: hwmon:
 lenovo,thinkpad-t14s-fan-controller: Add binding

Create a DT binding for the Lenovo Thinkpad T14s Gen6 Snapdragon fan
controller.

Signed-off-by: Sebastian Reichel <sre@kernel.org>
---
 .../lenovo,thinkpad-t14s-fan-controller.yaml  | 40 +++++++++++++++++++
 1 file changed, 40 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/hwmon/lenovo,thinkpad-t14s-fan-controller.yaml

diff --git a/Documentation/devicetree/bindings/hwmon/lenovo,thinkpad-t14s-fan-controller.yaml b/Documentation/devicetree/bindings/hwmon/lenovo,thinkpad-t14s-fan-controller.yaml
new file mode 100644
index 000000000..112d94910
--- /dev/null
+++ b/Documentation/devicetree/bindings/hwmon/lenovo,thinkpad-t14s-fan-controller.yaml
@@ -0,0 +1,40 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/hwmon/lenovo,thinkpad-t14s-fan-controller.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Lenovo Thinkpad T14s Fan Controller
+
+maintainers:
+  - Sebastian Reichel <sre@kernel.org>
+
+description:
+  The Qualcomm Snapdragon-based Lenovo Thinkpad T14s has a custom Fan Controller,
+  which is not part of the Embedded Controller.
+
+properties:
+  compatible:
+    const: lenovo,thinkpad-t14s-fan-controller
+
+  reg:
+    const: 0x36
+
+required:
+  - compatible
+  - reg
+
+additionalProperties: false
+
+examples:
+  - |+
+    i2c {
+        #address-cells = <1>;
+        #size-cells = <0>;
+
+        fan-controller@36 {
+            compatible = "lenovo,thinkpad-t14s-fan-controller";
+            reg = <0x36>;
+        };
+    };
+...
-- 
2.43.0

